### ISSUE TITLE
Support for parallel AIOs

### ISSUE THREAD

**Issue by DJBenson** [Codegnosis/givtcp-battery-card#39]
I've recently had my single AIO upgraded to a parallel install and noticed that the card stopped working after I did a bulk find/replace of my original AIO entities for the gateway versions. If I specifically pick the two separate batteries the card works (which is fine for a workaround) but if you selct the gateway serial the card is blank and I can't see any error messages in the debug console in Chrome.

Running latest (November) release of HA and the battery card is up to date too.

**Comment by Codegnosis** [comment]
Hi - Does it work if you use one card per AIO instance? The card was initially designed to support one inverter per card, but if there is enough demand I can see if it's feasible to extend that functionality to parallel systems (although can't promise anything 😊)

**Comment by DJBenson** [comment]
Hey thanks for the response. 

Yes it works if you add an inverter per card. The parallel AIO setup exposes three entities, one per inverter (which both work) and a combined entity from the gateway - this shows up in the card config but doesn't work.

It's a nice to have but if you do get round to it I'm happy to help test it. Only down side at the moment is taking up twice as much dashboard space with the two cards but first world problems I guess 🤣

**Comment by Codegnosis** [comment]
Hi @DJBenson - I've released v0.3.0-beta.6 (https://github.com/Codegnosis/givtcp-battery-card/releases/tag/v0.3.0-beta.6), which addresses a different issue but **_may_** also work for your environment, as it will allow you to define the combined entity IDs you mentioned.

The functionality can be toggled on/off using the editor (or YAML config).

If a custom sensor entity ID is detected by the card, it will:

- check to see if the sensor exists. If it cannot be found, it display the familiar "sensor not found" message
- use the custom defined sensor in place of the automatically derived sensor

Please have a go and see if it works for you, Thanks 🙏

**Comment by DJBenson** [comment]
Hi @Codegnosis 

I actually looked a bit more into this today and noticed that the gateway doesn't have all the required entities - it is missing the charge and discharge entities (which is precisely what the error was/is saying). 

I figured out a workaround which gets the card to work as I wanted by creating two helpers which are named in the expected format (i.e. sensor.gwXXXXXXXX_charge_power and sensor.gwXXXXXXXX_discharge_power which is the sum (using the "Combine the state of multiple sensors" helper) with the two individual sensors for each from the two batteries, i.e.

- sensor.givtcp_gwXXXXXXXX_charge_power:
  - sensor.givtcp2_chXXXXXXXX_charge_power
  - sensor.givtcp3_chXXXXXXXX_charge_power
- sensor.gwXXXXXXXX_discharge_power:
  - sensor.givtcp2_chXXXXXXXX_discharge_power
  - sensor.givtcp3_chXXXXXXXX_discharge_power

Example;
<img width="590" alt="Image" src="https://github.com/user-attachments/assets/4898347f-e005-4b5a-9a54-6d1c5845cdba" />

Result;
<img width="1014" alt="Image" src="https://github.com/user-attachments/assets/21870bb5-6f88-4b5e-84c3-25755fdfa5da" />

No special entity mapping required - once the helpers are created, the card springs to life.

Sorry for wasting your time on this but also thanks for inadvertently helping me to the solution 😀

**Comment by Codegnosis** [comment]
That's fantastic! Thanks for the additional information too - I can add a section to the readme covering this scenario. 🙏

**Comment by DJBenson** [comment]
There's another way it can be done but still requires some form of template - there is an entity on the gateway called sensor.givtcp_gwxxxxxxxx_invertor_power which is a +/- sensor depending on whether the inverter is charging or discharging - you could probably create two template sensors based on this sensor being <> 0 but I prefer the sum method described above.

**Comment by gcoan** [comment]
@Codegnosis would it be possible for the card to accept multiple sensor values from the individual AIO battery charge/sensors?

This would mean that people wouldn't have to create such template sensors, which to be honest are a bit of a cludge and fill the database up with lots of state changes (you'll get a state change whenever either of the underlying sensors change)

**Comment by Codegnosis** [comment]
> [@Codegnosis](https://github.com/Codegnosis) would it be possible for the card to accept multiple sensor values from the individual AIO battery charge/sensors?
> 
> This would mean that people wouldn't have to create such template sensors, which to be honest are a bit of a cludge and fill the database up with lots of state changes (you'll get a state change whenever either of the underlying sensors change)

Yes, I think this could be possible. I think it would perhaps require some changes to allow arrays of sensor entity IDs in the custom config, which would then be summed before the card runs its own calculations.

Can you open a new Feature Request issue and add some examples? I'll certainly look into it for the next release (it might take a while before I can get around to implementing it though). Thanks 🙏😊

**Comment by DJBenson** [comment]
Raised #48 for this.
